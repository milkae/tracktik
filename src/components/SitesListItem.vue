<template>
  <div class="sites-list__item">
    <div class="sites-list__item__image">
      <img :src="imageUrl" />
    </div>
    <div class="sites-list__item__info">
      <div class="sites-list__item__info__title">
        {{ site.title }}
      </div>
      <div class="sites-list__item__info__address">
        <p>{{ address.street }}</p>
        <p>{{ address.zipCode }} {{ address.city }}</p>
        <p>{{ address.state }}, {{ address.country }}</p>
      </div>
      <div class="sites-list__item__info__contact">
        Main contact: {{ mainContactName }}
      </div>
    </div>
    <router-link :to="url" class="sites-list__item__link"
      ><span class="lnr lnr-chevron-right"></span>
    </router-link>
  </div>
</template>

<script lang="ts">
import { Vue, Component, Prop } from "vue-property-decorator";
import { Site, SiteContact } from "@/store/sites/types";

@Component
export default class SitesListItem extends Vue {
  @Prop({ type: Object, required: true }) readonly site!: Site;
  get imageUrl(): string {
    return this.site.images[0];
  }
  get mainContact(): SiteContact {
    return this.site.contacts.main;
  }
  get mainContactName(): string {
    return this.mainContact.firstName + " " + this.mainContact.lastName;
  }
  get url(): string {
    return `site/${this.site.id}`;
  }

  get address(): string {
    return this.site.address;
  }
}
</script>

<style scoped lang="scss">
.sites-list__item {
  display: flex;
  align-items: center;
  padding: 0.5rem;
  border-top: 1px solid grey;

  &__image {
    margin-right: 0.5rem;
    overflow: hidden;
    border-radius: 50%;
    height: 20vw;
    width: 20vw;
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: grey;

    img {
      border-radius: 50%;
      width: auto;
      height: 100%;
    }
  }

  &__info {
    text-align: left;

    &__address {
      font-size: 0.8rem;
      color: grey;
      margin: 0.5rem 0;

      p {
        margin: 0;
        padding: 0;
      }
    }

    &__contact {
      font-size: 0.8rem;
    }
  }

  &__link {
    margin-left: auto;
    padding: 1rem;
    text-decoration: none;
    font-size: 1.5rem;
  }
}
</style>
