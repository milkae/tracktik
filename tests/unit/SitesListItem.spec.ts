import { shallowMount } from "@vue/test-utils";
import SitesListItem from "@/components/SitesListItem.vue";
import site from "./fixtures/site";

describe("SitesListItem", () => {
  let props: any;

  const build = () => {
    const wrapper = shallowMount(SitesListItem, {
      propsData: props,
      stubs: ["router-link"]
    });
    return {
      wrapper,
      sitesListItemImage: () => wrapper.find(".sites-list__item__image"),
      sitesListItemTitle: () => wrapper.find(".sites-list__item__title"),
      sitesListItemContact: () => wrapper.find(".sites-list__item__contact"),
      sitesListItemLink: () => wrapper.find(".sites-list__item__link")
    };
  };

  beforeEach(() => {
    props = {
      site
    };
  });

  it("renders the component", () => {
    const { wrapper } = build();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders main components", () => {
    const {
      sitesListItemImage,
      sitesListItemTitle,
      sitesListItemContact,
      sitesListItemLink
    } = build();

    const {
      site: { images, title, contacts, id }
    } = props;

    expect(sitesListItemImage().exists()).toBe(true);
    expect(sitesListItemImage().attributes().src).toBe(images[0]);

    expect(sitesListItemTitle().exists()).toBe(true);
    expect(sitesListItemTitle().text()).toBe(title);

    expect(sitesListItemContact().exists()).toBe(true);
    expect(sitesListItemContact().text()).toBe(
      contacts.main.firstname + " " + contacts.main.lastname
    );

    expect(sitesListItemLink().exists()).toBe(true);
    expect(sitesListItemLink().attributes().to).toBe(`site/${id}`);
  });
});
