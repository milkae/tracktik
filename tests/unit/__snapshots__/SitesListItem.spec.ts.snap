// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SitesListItem renders the component 1`] = `
<div class="sites-list__item"><img src="http://lorempixel.com/640/480/city" class="sites-list__item__image">
  <div class="sites-list__item__title">
    Kids Tower
  </div>
  <div class="sites-list__item__contact">
    undefined undefined
  </div>
  <router-link-stub to="site/ltuq30QviB" class="sites-list__item__link">&gt;</router-link-stub>
</div>
`;
